org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractFoldingTest.xtend

```diff
@@ -0,0 +1,115 @@
+/*******************************************************************************
+ * Copyright (c) 2019 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ui.testing
+
+import com.google.common.annotations.Beta
+import com.google.inject.Inject
+import org.eclipse.core.resources.IFile
+import org.eclipse.xtext.resource.FileExtensionProvider
+import org.eclipse.xtext.ui.XtextProjectHelper
+import org.eclipse.xtext.ui.editor.folding.IFoldingRegionProvider
+import org.eclipse.xtext.ui.editor.model.IXtextDocument
+import org.eclipse.xtext.ui.testing.AbstractEditorTest
+import org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil
+
+import static extension java.util.regex.Pattern.quote
+import static extension org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil.addNature
+
+/**
+ * @author miklossy - Initial contribution and API
+ * @since 2.19
+ */
+@Beta
+abstract class AbstractFoldingTest extends AbstractEditorTest {
+
+	@Inject protected extension FileExtensionProvider
+	@Inject protected extension IFoldingRegionProvider
+
+	/**
+	 * Special symbol indicating the beginning of a folding region.
+	 */
+	def String b() '''[>'''
+
+	/**
+	 * Special symbol indicating the end of a folding region.
+	 */
+	def String e() '''<]'''
+
+	/**
+	 * Test that the expected folding regions are present on a given DSL text.
+	 *
+	 * @param it The DSL text where the expected folding regions are enclosed by the {@link #b b} and {@link #e e} symbols.
```

Comment:
Please fix the method references in the JavaDoc here, too (should be `#b()`)

Reply:
done